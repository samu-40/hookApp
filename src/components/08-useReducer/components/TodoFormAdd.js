import React from 'react';
import PropTypes from 'prop-types';
import { useForm } from '../../../hooks/useForm';

const TodoFormAdd = ({handleAdd}) => {

    const [{description}, handleInputChanges, reset] = useForm({description:''});

    const handleSubmit = (e) => {
        e.preventDefault();

        if (description.trim().length <= 1) {return};

        const newTodo = {
            id: new Date().getTime(),
            desc: description,
            done: false,
        };

        handleAdd(newTodo);

        reset();

    };


    return (
        <>
            <h6>Agregar TODO</h6>
            <hr />

            <form
                className='d-grid gap-2'
                onSubmit={handleSubmit}
            >
                <input
                    type='text'
                    name='description'
                    placeholder='TODO....'
                    autoComplete='off'
                    className='form-control'
                    value={description}
                    onChange={handleInputChanges}
                />

                <button
                    type='submit'
                    className='btn btn-outline-success mt-1 '
                >
                    Add
                        </button>
            </form>
        </>
    );
};

TodoFormAdd.propTypes = {
    handleAdd: PropTypes.func.isRequired,
}

export default TodoFormAdd
