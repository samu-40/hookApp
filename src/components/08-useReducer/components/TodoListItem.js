import React from 'react';
import PropTypes from 'prop-types';

export const TodoListItem = ({todo, handleDelete, handleToggleDone, index}) => {

    return (
        <li
            className='list-group-item'
        >

            <p
                className={`${todo.done && 'complete'}`}
                onClick={() => handleToggleDone(todo.id)}
            >
                {index + 1}. {todo.desc}
            </p>

            <button
                className='btn btn-danger'
                onClick={() => handleDelete(todo.id)}
            >
                Delete
            </button>

        </li>
    );

};

TodoListItem.propTypes = {
    todo: PropTypes.object.isRequired,
    handleDelete: PropTypes.func.isRequired,
    handleToggleDone: PropTypes.func.isRequired,
    index: PropTypes.number.isRequired,
};
