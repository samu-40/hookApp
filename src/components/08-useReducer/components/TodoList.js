import React from 'react'
import { TodoListItem } from './TodoListItem'
import PropTypes from 'prop-types';

export const TodoList = ({todos, handleDelete, handleToggleDone}) => {

    return (
        <ul>
            {
                todos.map((todo, i) => (
                    <TodoListItem 
                        key={todo.id}
                        todo={todo}
                        handleDelete={handleDelete}
                        handleToggleDone={handleToggleDone}
                        index={i}
                    />
                ))
            }
        </ul>
    );

};

TodoListItem.propTypes = {
    todos: PropTypes.array,
    handleDelete: PropTypes.func.isRequired,
    handleToggleDone: PropTypes.func.isRequired,
};
