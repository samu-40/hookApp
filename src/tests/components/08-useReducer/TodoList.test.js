import {shallow} from 'enzyme';
import { TodoList } from '../../../components/08-useReducer/components/TodoList';
import { demoTodos } from '../../fixtures/demoTodos';

describe('Pruebas en TodoList.js', () => {

    const wrapper = shallow(
        <TodoList 
            todos = {demoTodos}
            handleDelete = {()=>{}}
            handleToggleDone = {()=>{}}
        />
    );

    test('Mostrat el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Wrapper debe tener 2 <TodoListItem/>', () => {
        expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length);
    });

    test('TodoListItem debe tener sus funciones como props', () => {
        const fnc1 = (wrapper.find('TodoListItem').at(0).prop('handleDelete'));
        const fnc2 = (wrapper.find('TodoListItem').at(0).prop('handleToggleDone'));

        expect(fnc1).toEqual(expect.any(Function));
        expect(fnc2).toEqual(expect.any(Function));
    });
    
});
