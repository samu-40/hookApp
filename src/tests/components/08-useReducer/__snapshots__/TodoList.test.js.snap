// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en TodoList.js Mostrat el componente correctamente 1`] = `
<ul>
  <TodoListItem
    handleDelete={[Function]}
    handleToggleDone={[Function]}
    index={0}
    key="1"
    todo={
      Object {
        "desc": "Aprender pintura",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[Function]}
    handleToggleDone={[Function]}
    index={1}
    key="2"
    todo={
      Object {
        "desc": "Aprender beisbol",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
